### PR TITLE
prosody: add LDAP(S) authentication via SASL mechanism

### PR DESCRIPTION
**Pull request by netaskd**
PR adds ldap authentication via sasl mechanism
started from #50

### CODE REVIEW

**Comment by saghul** [comment]
@netaskd This looks good to me. I trust that it works, so I hope to merge it this week after a bit of testing just to check if it adds no regressions. Cheers!

**Comment by deja-geek** [comment]
Fresh build on ubuntu 18.04 LTS.  Getting this error in the logs:

> {"log":"Error: /config/conf.d/jitsi-meet.cfg.lua:5: unexpected symbol near '{'\n","stream":"stdout","time":"2019-03-15T22:48:11.508900611Z"}

**Comment by netaskd** [comment]
@deja-geek 
I guess it's not about this PR. Looks like about jwt
https://github.com/jitsi/docker-jitsi-meet/blob/78e05dc494d262c967ffb641f6cb1ea17767766b/prosody/rootfs/defaults/conf.d/jitsi-meet.cfg.lua#L5

**Comment by h3po** [comment]
I just built the prosody image from pr_sasl_ldap and authentication works against active directory. IMO this is missing an LDAP_TLS_CACERT_DIR variable in order to use LDAP_TLS_CHECK_PEER.

Is there a way to get username or email address from ldap for authenticated users? Although logged in, everyone is named "fellow jitster" in my test room.

File: prosody/rootfs/etc/cont-init.d/10-config [hunk 1]
```diff
@@ -1,5 +1,13 @@
 #!/usr/bin/with-contenv bash
 
+if [[ ! -f /config/saslauthd.conf ]]; then
+  cp /defaults/saslauthd.conf /config/
+  tpl /config/saslauthd.conf > /etc/saslauthd.conf
```
Comment (jkroepke) [inline prosody/rootfs/etc/cont-init.d/10-config]: If I restart the container the `/config/saslauthd.conf` does exists while `/etc/saslauthd.conf` not.

Looks like u have to split the condition here.

Comment (netaskd) [reply prosody/rootfs/etc/cont-init.d/10-config]: Hi @jkroepke.
Fixed, try to use d336f39

Comment (netaskd) [reply prosody/rootfs/etc/cont-init.d/10-config]: rebased in 14996e0

**Comment by netaskd** [comment]
Hi @h3po,
1. I've added two vars LDAP_TLS_CACERT_DIR and LDAP_TLS_CACERT_FILE in latest commit.
2. I don't think that it's possible right now. Maybe @saghul can say more about it.

**Comment by saghul** [comment]
> Is there a way to get username or email address from ldap for authenticated users? Although logged in, everyone is named "fellow jitster" in my test room.

TBH I'm not sure that's possible.

**Comment by deja-geek** [comment]
> I just built the prosody image from pr_sasl_ldap and authentication works against active directory. 

What was the configuration you used to get authentication against LDAP?

**Comment by jkroepke** [comment]
> TBH I'm not sure that's possible.

I'don't know if its possible with `ldap2` (https://modules.prosody.im/mod_auth_ldap2.html) but its an community module.

I test this branch agist a real world LDAP (openldap) and the patch works fine. Users can create room only if they authenticated via ldap.

Hopefully it will be merged soon.

**Comment by h3po** [comment]
@saghul Do you think you'll have time to look at and merge this soon?

**Comment by saghul** [comment]
Fantastic work @netaskd! Landed on dev.

**Pull request merged**